src/packages/BooleanQueryTokenizer/BooleanQueryTokenizer.ts

```diff
@@ -148,6 +155,94 @@ export class BooleanQueryTokenizer {
+                let j = i + 1;
+                while (j < tokens.length) {
+                    const lookAheadToken = tokens[j];
+                    if (lookAheadToken.type === TokenType.separator) {
+                        j += 1;
+                    } else if (BooleanQueryTokenizer.isNegationable(lookAheadToken)) {
+                        BooleanQueryTokenizer.replaceTokens(tokens, i, j);
+                        break;
+                    } else if (BooleanQueryTokenizer.isParenthesisGroup(lookAheadToken)) {
+                        let parenthesisTracker = 0;
+                        for (let k = j; k < tokens.length; k += 1) {
+                            const parenthesisLoopToken = tokens[k];
+                            if (parenthesisLoopToken.type === TokenType.parentheses) {
+                                if (parenthesisLoopToken.lexeme === '(') {
+                                    parenthesisTracker += 1;
+                                } else if (parenthesisLoopToken.lexeme === ')') {
+                                    parenthesisTracker -= 1;
+                                }
+                            }
+                            if (parenthesisTracker === 0) {
+                                BooleanQueryTokenizer.replaceTokens(tokens, i, k);
+                                break;
+                            }
+                        }
+                        break;
+                    } else {
+                        break;
+                    }
+                }
+            }
+        }
+        // Filter only words and phrases
+        const wordsAndPhrases = tokens
+            .filter((token) => token.type === TokenType.word || token.type === TokenType.phrase)
+            .map((wordOrPhrase) => {
+                // `phrase` tokens require some dirty preprocessing
+                // to remove quotes and possible inner wildcards.
+                if (wordOrPhrase.type === TokenType.phrase) {
+                    return wordOrPhrase.lexeme.replace(/('|"|\*)/g, '').trim();
+                }
+                return wordOrPhrase.lexeme;
+            })
+            // After `phrase` tokens preprocessing, some strings can become empty,
+            // so we should remove them.
+            .filter((str) => str);
+        return {
+            wordsAndPhrases,
+        };
+    }
+
+    private static replaceTokens(tokens: Token[], from: number, to: number): Token[] {
+        tokens.forEach((_token, index) => {
+            if (index >= from && index <= to) {
+                // eslint-disable-next-line no-param-reassign
+                tokens[index] = {
+                    type: TokenType.unknown,
+                    lexeme: '',
+                };
+            }
+        });
+        return tokens;
+    }
+
+    private static isNegation(token: Token): boolean {
+        return (
+            (token.type === TokenType.booleanOperator &&
```

Comment:
I understand that prettier configured to have such code style, but for me looks hard to read.
Something like this is much more obvious:
```suggestion
            (
                token.type === TokenType.booleanOperator
                && (token.lexeme === 'NOT' || token.lexeme === 'AND NOT')
            )
            || (
                token.type === TokenType.modifier
                && (token.lexeme === '-' || token.lexeme === '!' )
            )
```

or

```suggestion
            (token.type === TokenType.booleanOperator
                && (token.lexeme === 'NOT' || token.lexeme === 'AND NOT'))
            || (token.type === TokenType.modifier
                && (token.lexeme === '-' || token.lexeme === '!' ))
```
---
This is not remark to current PR but just point to think about code style rules

Reply:
I also struggled with prettifier, doesn't make much sense to me, but let's change it in a separate MR

Created a task https://jira.textkernel.nl/browse/ONEUI-235